src/resources/auth.html

```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Authentication required</title>
+    <script src="/resources/jquery-3.1.1.min.js"></script>
+
+    <script type="application/javascript">
+      function getQueryVariable(variable) {
+        var query = window.location.search.substring(1);
+        var vars = query.split('&');
+        for (var i = 0; i < vars.length; i++) {
+          var pair = vars[i].split('=');
+          if (decodeURIComponent(pair[0]) == variable) {
+            return decodeURIComponent(pair[1]);
+          }
+        }
+      }
+
+      window.onload = function() {
+        var redirect_uri = getQueryVariable('redirect_uri');
+
+        if (redirect_uri === undefined || redirect_uri === null) {
+          $('#notAuthorizedDiv').css("display", "block");
+        } else {
+          $('#authenticationRequiredDiv').css("display", "block");
+
+          // Allow redirect only to Github
+          if (redirect_uri.indexOf('https://github.com/login/oauth/authorize') === 0) {
+            $('#authenticationButton').click(function () {
```

Comment:
you don't need jquery for that, just `document.etElementById('authenticationButton').addEventListener('click', function() { /* ... */ }, false);`